kani-compiler/src/intrinsics.rs

```diff
@@ -471,55 +466,55 @@ impl Intrinsic {
 fn try_match_atomic(intrinsic_instance: &Instance) -> Option<Intrinsic> {
     let intrinsic_str = intrinsic_instance.intrinsic_name().unwrap();
     let sig = intrinsic_instance.ty().kind().fn_sig().unwrap().skip_binder();
-    if let Some(suffix) = intrinsic_str.strip_prefix("atomic_and_") {
+    if let Some(suffix) = intrinsic_str.strip_prefix("atomic_and") {
```

Comment:
In a previous instance of a similar change, I changed this to be an equality check (as opposed to a prefix one): https://github.com/model-checking/kani/pull/4123/files

I'm OK either way though.